File: 2439-minimize-maximum-of-array/2439-minimize-maximum-of-array.cpp
#define ll long long

class Solution {
public:
    int minimizeArrayValue(vector<int>& nums) {
        int ans = 0;
        ll prefix_sum = 0;
        for (int i = 0; i < nums.size(); ++i){
            prefix_sum += nums[i];
            int curr = ceil((double)prefix_sum / (i + 1));
            ans = max(ans, curr);
        }
        return ans;
    }
};

File: 2439-minimize-maximum-of-array/2439-minimize-maximum-of-array.py
class Solution:
    def minimizeArrayValue(self, nums: List[int]) -> int:
        ans, prefix_sum = 0, 0
        
        for i in range(len(nums)):
            prefix_sum += nums[i]
            ans = max(ans, ceil(prefix_sum / (i + 1)))
        
        return ans